internal/filter/filter.go

```diff
@@ -27,6 +31,34 @@ var skipDeletions predicate.Predicate = predicate.Funcs{
 	DeleteFunc: func(_ event.DeleteEvent) bool { return false },
 }
 
+var kmmClusterClaimChanged predicate.Predicate = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		if e.ObjectOld == nil {
```

Comment:
We don't need to check if the New and Old objects are nil, in the update Event they cannot be nil, i think 
we need to check that the casting is OK

Reply:
Thank you for the comment! Updated accordingly.